sidechain/consensus/aura/src/lib.rs

```diff
@@ -455,15 +474,83 @@ mod tests {
+		let slot_info = now_slot(2.into(), &latest_parentchain_header);
+
+		let result = SimpleSlotWorker::on_slot(&mut aura, slot_info, Default::default());
+
+		assert!(result.is_none());
+		assert!(!parentchain_block_import_trigger.has_import_been_called());
+	}
+
+	#[test]
+	fn on_slot_claims_slot_if_latest_parentchain_header_in_queue_contains_correspondent_validateer_set(
+	) {
+		let _ = env_logger::builder().is_test(true).try_init();
+		let already_imported_parentchain_header =
+			ParentchainHeaderBuilder::default().with_number(84).build();
+		let latest_parentchain_header = ParentchainHeaderBuilder::default().with_number(85).build();
+		let parentchain_block_import_trigger =
+			create_import_trigger_with_header(latest_parentchain_header.clone());
+		let validateer_set_one = create_validateer_set_from_publics(vec![
+			Keyring::Alice.public(),
+			Keyring::Bob.public(),
+		]);
+		let validateer_set_two = create_validateer_set_from_publics(vec![
+			Keyring::Alice.public(),
+			Keyring::Bob.public(),
+			Keyring::Charlie.public(),
+		]);
+		let onchain_mock = OnchainMock::default()
+			.add_validateer_set(&already_imported_parentchain_header, Some(validateer_set_one))
+			.add_validateer_set(&latest_parentchain_header, Some(validateer_set_two));
+
+		let mut aura = get_aura(onchain_mock, parentchain_block_import_trigger.clone());
+
+		let slot_info = now_slot(3.into(), &already_imported_parentchain_header);
+
+		let result = SimpleSlotWorker::on_slot(&mut aura, slot_info, Default::default()).unwrap();
 
+		assert_eq!(
+			result.block.block.block_data().layer_one_head,
+			latest_parentchain_header.hash()
+		);
+		assert!(parentchain_block_import_trigger.has_import_been_called());
+	}
+
+	#[test]
+	fn on_slot_does_not_claim_slot_if_latest_parentchain_header_in_queue_contains_correspondent_validateer_set(
+	) {
+		let _ = env_logger::builder().is_test(true).try_init();
+		let already_imported_parentchain_header =
+			ParentchainHeaderBuilder::default().with_number(84).build();
+		let latest_parentchain_header = ParentchainHeaderBuilder::default().with_number(85).build();
+		let parentchain_block_import_trigger =
+			create_import_trigger_with_header(latest_parentchain_header.clone());
+		let validateer_set_one = create_validateer_set_from_publics(vec![
+			Keyring::Alice.public(),
+			Keyring::Bob.public(),
+		]);
+		let validateer_set_two = create_validateer_set_from_publics(vec![
+			Keyring::Alice.public(),
+			Keyring::Bob.public(),
+			Keyring::Charlie.public(),
+		]);
+		let onchain_mock = OnchainMock::default()
+			.add_validateer_set(&already_imported_parentchain_header, Some(validateer_set_one))
+			.add_validateer_set(&latest_parentchain_header, Some(validateer_set_two));
+
+		let mut aura = get_aura(onchain_mock, parentchain_block_import_trigger.clone());
+
+		// If the validateer set one (instead of the latest one) is looked up, the slot will be claimed. But it should not, as the latest one should be used.
```

Comment:
From the code I can see that the slot is not claimed. But what does this part of the comment mean then?

```
If the validateer set one (instead of the latest one) is looked up, the slot will be claimed
```

Reply:
It should make clear, that if the `already_imported_parentchain_header` is used (instead of the `latest_parentchain_header`, which will be used in the sidechain block), then the slot is claimed (which was previously the case). So this would have failed with the previous slot claiming strategy.

Reply:
Ok, I mean I can understand it, but from outside it's still a bit obscure. May I ask for a more concise rephrase? :)

Something like:
```
// Does not claim the slot because it's not our turn with the next validateer set of the queued block
```